legal-api/src/legal_api/services/search_service.py

```diff
@@ -0,0 +1,265 @@
+# Copyright © 2025 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This provides the service for getting business details as of a filing."""
+# pylint: disable=singleton-comparison ; pylint does not recognize sqlalchemy ==
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from operator import and_
+from typing import Final, List, Optional, Tuple
+
+from requests import Request
+from sqlalchemy import func
+
+from legal_api.models import Business, Filing, db
+
+
+@dataclass
+class AffiliationSearchDetails:  # pylint: disable=too-many-instance-attributes
+    """Used for filtering Identifiers based on filters passed."""
+
+    identifier: Optional[str]
+    status: Optional[List[str]]
+    name: Optional[str]
+    type: Optional[List[str]]
+    page: int
+    limit: int
+
+    @classmethod
+    def from_request_args(cls, req: Request):
+        """Create an instance from request arguments."""
+        return cls(
+            identifier=req.get('identifier', None),
+            name=req.get('name', None),
+            type=req.get('type', []),
+            status=req.get('status', []),
+            page=int(req.get('page', 1)),
+            limit=int(req.get('limit', 100000))
```

Comment:
This is to support existing calls that don't include the "limit" parameter, right?

Does this need to be updated eventually? When? Would this affect API users?